CI is failing with:
```
 Caused by: java.lang.RuntimeException: Failed to register resource META-INF/services/org.jboss.logmanager.MDCProvider
	at io.quarkus.runtime.ResourceHelper.lambda$registerResources$0(ResourceHelper.java:37)
	at io.quarkus.runtime.util.ClassPathUtils.lambda$consumeStream$1(ClassPathUtils.java:180)
	at io.quarkus.runtime.util.ClassPathUtils.readStream(ClassPathUtils.java:201)
	at io.quarkus.runtime.util.ClassPathUtils.consumeStream(ClassPathUtils.java:179)
	at io.quarkus.runtime.util.ClassPathUtils.consumeAsStreams(ClassPathUtils.java:71)
	at io.quarkus.runtime.ResourceHelper.registerResources(ResourceHelper.java:33)
	... 9 more
Caused by: java.lang.IllegalAccessException: class io.quarkus.runtime.ResourceHelper cannot access class com.oracle.svm.core.jdk.Resources (in module org.graalvm.nativeimage.builder) because module org.graalvm.nativeimage.builder does not export com.oracle.svm.core.jdk to unnamed module @3b576ee3
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:392)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674)
	at java.base/java.lang.reflect.Method.invoke(Method.java:560)
	at io.quarkus.runtime.ResourceHelper.lambda$registerResources$0(ResourceHelper.java:35)
	... 14 more
```

--- a/.github/workflows/ci-actions-incremental.yml
+++ b/.github/workflows/ci-actions-incremental.yml
@@ -712,10 +712,13 @@ jobs:
         if: startsWith(matrix.os-name, 'windows')
       - name: Setup GraalVM
         id: setup-graalvm
-        uses: DeLaGuardo/setup-graalvm@master
+        uses: graalvm/setup-graalvm@v1
         if: startsWith(matrix.os-name, 'windows')
         with:
-          graalvm-version: '21.3.0.java11'
+          version: 'latest'
+          java-version: '17'
+          components: 'native-image'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Install native-image component
         if: startsWith(matrix.os-name, 'windows')
         run: |

--- a/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/JPMSExportBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/JPMSExportBuildItem.java
@@ -16,7 +16,7 @@ public final class JPMSExportBuildItem extends MultiBuildItem {
     private final GraalVM.Version exportBefore;
 
     public JPMSExportBuildItem(String moduleName, String packageName) {
-        this(moduleName, packageName, GraalVM.Version.MINIMUM, null);
+        this(moduleName, packageName, null, null);
     }
 
     public JPMSExportBuildItem(String moduleName, String packageName, GraalVM.Version exportAfter) {
@@ -62,5 +62,10 @@ public final class JPMSExportBuildItem extends MultiBuildItem {
 
     public GraalVM.Version getExportBefore() {
         return exportBefore;
+    }
+
+    public boolean isRequired(GraalVM.Version current) {
+        return (exportAfter == null || current.compareTo(exportAfter) > 0) &&
+                (exportBefore == null || current.compareTo(exportBefore) < 0);
     }
 }

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java
@@ -23,14 +23,10 @@ public final class GraalVM {
         static final Version UNVERSIONED = new Version("Undefined", "snapshot", Distribution.ORACLE);
         static final Version VERSION_21_3 = new Version("GraalVM 21.3", "21.3", Distribution.ORACLE);
         static final Version VERSION_21_3_0 = new Version("GraalVM 21.3.0", "21.3.0", Distribution.ORACLE);
-        static final Version VERSION_21_3_2 = new Version("GraalVM 21.3.2", "21.3.2", Distribution.ORACLE);
-        static final Version VERSION_22_0_0_2 = new Version("GraalVM 22.0.0.2", "22.0.0.2", Distribution.ORACLE);
         public static final Version VERSION_22_3_0 = new Version("GraalVM 22.3.0", "22.3.0", Distribution.ORACLE);
+        public static final Version VERSION_22_2_0 = new Version("GraalVM 22.2.0", "22.2.0", Distribution.ORACLE);
 
-        public static final Version VERSION_22_1_0 = new Version("GraalVM 22.1.0", "22.1.0", Distribution.ORACLE);
-        static final Version VERSION_22_2_0 = new Version("GraalVM 22.2.0", "22.2.0", Distribution.ORACLE);
-
-        public static final Version MINIMUM = VERSION_21_3;
+        public static final Version MINIMUM = VERSION_22_2_0;
         public static final Version CURRENT = VERSION_22_2_0;
         public static final int UNDEFINED = -1;
 

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -82,8 +82,7 @@ public class NativeImageBuildStep {
     @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
     void addExportsToNativeImage(BuildProducer<JPMSExportBuildItem> exports) {
         // Needed by io.quarkus.runtime.ResourceHelper.registerResources
-        exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk",
-                GraalVM.Version.VERSION_22_1_0));
+        exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk"));
     }
 
     @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
@@ -720,11 +719,7 @@ public class NativeImageBuildStep {
                 nativeImageArgs.add("-J--add-opens=java.base/java.util=ALL-UNNAMED");
 
                 if (nativeConfig.enableReports) {
-                    if (graalVMVersion.isOlderThan(GraalVM.Version.VERSION_21_3_2)) {
-                        nativeImageArgs.add("-H:+PrintAnalysisCallTree");
-                    } else {
-                        nativeImageArgs.add("-H:PrintAnalysisCallTreeType=CSV");
-                    }
+                    nativeImageArgs.add("-H:PrintAnalysisCallTreeType=CSV");
                 }
 
                 // only available in GraalVM 22.3.0 and better.
@@ -757,12 +752,8 @@ public class NativeImageBuildStep {
                     nativeImageArgs.add("--no-fallback");
                 }
 
-                // 22.1 removes --allow-incomplete-classpath and makes it the default. --link-at-build-time is now
-                // needed to bring back the old behavior (which requires all classes to be resolvable at build time).
-                if (graalVMVersion.isNewerThan(GraalVM.Version.VERSION_22_0_0_2) && !classpathIsBroken) {
+                if (!classpathIsBroken) {
                     nativeImageArgs.add("--link-at-build-time");
-                } else if (!graalVMVersion.isNewerThan(GraalVM.Version.VERSION_22_0_0_2) && classpathIsBroken) {
-                    nativeImageArgs.add("--allow-incomplete-classpath");
                 }
 
                 if (nativeConfig.reportErrorsAtRuntime) {
@@ -904,11 +895,6 @@ public class NativeImageBuildStep {
                     nativeImageArgs.add("--exclude-config");
                     nativeImageArgs.add(excludeConfig.getJarFile());
                     nativeImageArgs.add(excludeConfig.getResourceName());
-                }
-
-                // Work around https://github.com/quarkusio/quarkus/issues/21372
-                if (graalVMVersion.is(GraalVM.Version.VERSION_21_3_0) && graalVMVersion.isJava17()) {
-                    nativeImageArgs.add("-J--add-exports=java.management/sun.management=ALL-UNNAMED");
                 }
 
                 nativeImageArgs.add(nativeImageName);

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java
@@ -852,9 +852,7 @@ public class NativeImageBuildStep {
                 if (jpmsExports != null) {
                     HashSet<JPMSExportBuildItem> deduplicatedJpmsExport = new HashSet<>(jpmsExports);
                     for (JPMSExportBuildItem jpmsExport : deduplicatedJpmsExport) {
-                        GraalVM.Version exportBeforeVersion = jpmsExport.getExportBefore();
-                        if (graalVMVersion.isNewerThan(jpmsExport.getExportAfter()) &&
-                                (exportBeforeVersion == null || graalVMVersion.isOlderThan(exportBeforeVersion))) {
+                        if (jpmsExport.isRequired(graalVMVersion)) {
                             nativeImageArgs.add(
                                     "-J--add-exports=" + jpmsExport.getModule() + "/" + jpmsExport.getPackage()
                                             + "=ALL-UNNAMED");

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java
@@ -131,12 +131,11 @@ public class NativeImageFeatureStep {
             List<LambdaCapturingTypeBuildItem> lambdaCapturingTypeBuildItems,
             List<NativeImageResourcePatternsBuildItem> resourcePatterns) {
         // required in order to access org.graalvm.nativeimage.impl.RuntimeSerializationSupport and org.graalvm.nativeimage.impl.ConfigurationCondition
-        features.produce(
-                new JPMSExportBuildItem("org.graalvm.sdk", "org.graalvm.nativeimage.impl", GraalVM.Version.VERSION_22_1_0));
+        features.produce(new JPMSExportBuildItem("org.graalvm.sdk", "org.graalvm.nativeimage.impl"));
         // required in order to access com.oracle.svm.core.jni.JNIRuntimeAccess in GraalVM 22.2.x
         if (jniRuntimeAccessibleClasses != null && !jniRuntimeAccessibleClasses.isEmpty()) {
             features.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jni",
-                    GraalVM.Version.VERSION_22_1_0, GraalVM.Version.VERSION_22_3_0));
+                    null, GraalVM.Version.VERSION_22_3_0));
         }
     }
 
@@ -290,7 +289,7 @@ public class NativeImageFeatureStep {
         if (!proxies.isEmpty()) {
             // Needed to access DYNAMIC_PROXY_REGISTRY in GraalVM 22.2
             exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk.proxy",
-                    GraalVM.Version.VERSION_22_1_0, GraalVM.Version.VERSION_22_3_0));
+                    null, GraalVM.Version.VERSION_22_3_0));
 
             ResultHandle versionCompareto22_3Result = overallCatch.invokeVirtualMethod(VERSION_COMPARE_TO,
                     overallCatch.invokeStaticMethod(VERSION_CURRENT),
@@ -332,7 +331,7 @@ public class NativeImageFeatureStep {
         if (!resourcePatterns.isEmpty()) {
             // Needed to access com.oracle.svm.core.configure.ResourcesRegistry.* in GraalVM 22.2
             exports.produce(new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.configure",
-                    GraalVM.Version.VERSION_22_1_0, GraalVM.Version.VERSION_22_3_0));
+                    null, GraalVM.Version.VERSION_22_3_0));
 
             TryBlock tc = overallCatch.tryBlock();
 
@@ -402,7 +401,7 @@ public class NativeImageFeatureStep {
             // Needed to access LOCALIZATION_FEATURE
             exports.produce(
                     new JPMSExportBuildItem("org.graalvm.nativeimage.builder", "com.oracle.svm.core.jdk.localization",
-                            GraalVM.Version.VERSION_22_1_0, GraalVM.Version.VERSION_22_3_0));
+                            null, GraalVM.Version.VERSION_22_3_0));
 
             BranchResult graalVm22_3Test = overallCatch.ifGreaterEqualZero(overallCatch.invokeVirtualMethod(VERSION_COMPARE_TO,
                     overallCatch.invokeStaticMethod(VERSION_CURRENT),

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/RegisterForReflectionITCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/RegisterForReflectionITCase.java
@@ -5,9 +5,6 @@ import static org.hamcrest.Matchers.startsWith;
 
 import org.junit.jupiter.api.Test;
 
-import io.quarkus.test.junit.DisableIfBuiltWithGraalVMNewerThan;
-import io.quarkus.test.junit.DisableIfBuiltWithGraalVMOlderThan;
-import io.quarkus.test.junit.GraalVMVersion;
 import io.quarkus.test.junit.QuarkusIntegrationTest;
 import io.restassured.RestAssured;
 
@@ -39,23 +36,12 @@ public class RegisterForReflectionITCase {
     }
 
     @Test
-    @DisableIfBuiltWithGraalVMOlderThan(GraalVMVersion.GRAALVM_22_1)
     public void testTargetWithNestedPost22_1() {
         final String resourceC = BASE_PKG + ".ResourceC";
 
         // Starting with GraalVM 22.1 ResourceC implicitly gets registered by GraalVM
         // (see https://github.com/oracle/graal/pull/4414)
         assertRegistration("ResourceC", resourceC);
-        assertRegistration("InaccessibleClassOfC", resourceC + "$InaccessibleClassOfC");
-        assertRegistration("OtherInaccessibleClassOfC", resourceC + "$InaccessibleClassOfC$OtherInaccessibleClassOfC");
-    }
-
-    @Test
-    @DisableIfBuiltWithGraalVMNewerThan(GraalVMVersion.GRAALVM_22_0)
-    public void testTargetWithNestedPre22_1() {
-        final String resourceC = BASE_PKG + ".ResourceC";
-
-        assertRegistration("FAILED", resourceC);
         assertRegistration("InaccessibleClassOfC", resourceC + "$InaccessibleClassOfC");
         assertRegistration("OtherInaccessibleClassOfC", resourceC + "$InaccessibleClassOfC$OtherInaccessibleClassOfC");
     }
@@ -70,7 +56,6 @@ public class RegisterForReflectionITCase {
     }
 
     @Test
-    @DisableIfBuiltWithGraalVMOlderThan(GraalVMVersion.GRAALVM_22_1)
     public void testLambdaCapturing() {
         final String resourceLambda = BASE_PKG + ".ResourceLambda";
 

